File: Cpp/2125_Number_of_Laser_Beams_in_a_Bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int result = 0;
        int row1 = 0;
        int row2 = 0;
        for(int i = 0; i < bank.size(); i++){
            int c = count(bank[i]);
            if(c > 0){
                row1 = row2;
                row2 = c;
                result += row1 * row2;
            }
        }
        return result;
    }
private:
    int count(string str){
        int c = 0;
        for(int i = 0; i < str.size(); i++){
            if(str[i] == '1'){
                c++;
            }
        }
        return c;
    }
};

File: CS/2125_Number_of_Laser_Beams_in_a_Bank.cs
public class Solution {
    public int NumberOfBeams(string[] bank) {
        int result = 0;
        int f1 = 0;
        int f2 = 0;
        for(int i = 0; i < bank.Length; i++){
            int temp = 0;
            for(int j = 0; j < bank[i].Length; j++){
                if(bank[i][j] == '1'){
                    temp++;
                }
            }
            if(temp!=0){
                f2 = f1;
                f1 = temp;
                result += f1*f2;
            }
        }
        return result;
    }
}
